modules/modules.cmake

```diff
@@ -0,0 +1,19 @@
+# SPDX-License-Identifier: Apache-2.0
+
+file(GLOB cmake_modules "${CMAKE_CURRENT_LIST_DIR}/*/CMakeLists.txt")
+
+foreach(module ${cmake_modules})
+  get_filename_component(module_dir  ${module} DIRECTORY)
+  get_filename_component(module_name ${module_dir} NAME)
+  string(TOUPPER ${module_name} MODULE_NAME_UPPER)
+  set(ZEPHYR_${MODULE_NAME_UPPER}_CMAKE_DIR ${module_dir})
+endforeach()
+
+file(GLOB kconfig_modules "${CMAKE_CURRENT_LIST_DIR}/*/Kconfig.*")
+
+foreach(module ${kconfig_modules})
+  get_filename_component(module_dir  ${module} DIRECTORY)
+  get_filename_component(module_name ${module_dir} NAME)
+  string(TOUPPER ${module_name} MODULE_NAME_UPPER)
+  set(ZEPHYR_${MODULE_NAME_UPPER}_KCONFIG ${module_dir}/Kconfig.${module_name})
```

Comment:
As the module name is to be the last component of the directory, why is it required that it appears also as the filename extension? Wouldn't just `Kconfig` be sufficient?

Reply:
good point, having it as suffix is redundant.

Just happened to follow how the `Kconfig.{module}` files were already named today here:
https://github.com/zephyrproject-rtos/zephyr/tree/master/modules

But when moving those files into own subdirs and located with corresponding CMakeLists.txt files, the suffix could be dropped.

Anyone opposing to dropping the module name suffix, please comment, otherwise I will follow the suggestion from @anangl .

Reply:
Done. Thanks for the suggestion.